weatherman.py

```diff
@@ -0,0 +1,246 @@
+__weather__ = {}  # for storing weather records
+__months__ = {1: "Jan", 2: "Feb", 3: "Mar", 4: "Apr", 5: "May", 6: "Jun", 7: "Jul", 8: "Aug", 9: "Sep", 10: "Oct",
+              11: "Nov", 12: "Dec"}
+
+import os
+import sys
+
+
+class Parser:
+
+    @staticmethod
+    def extracting_info_file_name(file_name):
+        city, weather, year, month = file_name.split("_")
+        city = city.split("/")[-1]
+        month, ext = month.split(".")
+        return city, year, month
+
+    def read_file(self, file_name):
+        city, year, month = self.extracting_info_file_name(file_name)
+        if city not in __weather__.keys():
+            __weather__[city] = {}
+        if year not in __weather__[city].keys():
+            __weather__[city][year] = {}
+        if month not in __weather__[city][year].keys():
+            __weather__[city][year][month] = {}
+        # reading file content
+        file = open(file_name, 'r')
+        lines = file.readlines()
+
+        for line in range(1, len(lines)):
+            info = lines[line].split(",")
+            details = {}
+            details.update({"PKT": info[0]})
+            details.update({"Max TemperatureC": int(info[1]) if info[1] != '' else None})
+            details.update({"Mean TemperatureC": int(info[2])if info[2] != '' else None})
+            details.update({"Min TemperatureC": int(info[3]) if info[3] != '' else None})
+            details.update({"Dew PointC": int(info[4]) if info[4] != '' else None})
+            details.update({"MeanDew PointC": int(info[5]) if info[5] != '' else None})
+            details.update({"Min DewpointC": int(info[6]) if info[6] != '' else None})
+            details.update({"Max Humidity": int(info[7]) if info[7] != '' else None})
+            details.update({"Mean Humidity": int(info[8]) if info[8] != '' else None})
+            details.update({"Min Humidity": int(info[9]) if info[9] != '' else None})
+            details.update({"Max Sea Level PressurehPa": info[10]})
+            details.update({"Mean Sea Level PressurehPa": info[11]})
+            details.update({"Min Sea Level PressurehPa": info[12]})
+            details.update({"Max VisibilityKm": info[13]})
+            details.update({"Mean VisibilityKm": info[14]})
+            details.update({"Min VisibilitykM": info[15]})
+            details.update({"Max Wind SpeedKm/h": info[16]})
+            details.update({"Max Gust SpeedKm/h": info[17]})
+            details.update({"Precipitationmm": info[18]})
+            details.update({"CloudCover": info[19]})
+            details.update({"Events": info[20]})
+            details.update({"WindDirDegrees": info[21]})
```

Comment:
Are we using all these keys? What is the purpose of storing and reading all of the irrelevant data?